File: Java/Tema3_Ejercicio6.java
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 6
 * =================================================
 * Escriba un programa que determine la categoría de edad de una persona.
 */

import java.util.Scanner;

public class Tema3_Ejercicio6 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite su edad: ");
        int edad = scanner.nextInt();
        
        System.out.println("\n=== CATEGORÍA DE EDAD ===");
        
        String categoria;
        if (edad < 0) {
            categoria = "Edad inválida";
        } else if (edad < 5) {
            categoria = "Bebé";
        } else if (edad < 12) {
            categoria = "Niño";
        } else if (edad < 18) {
            categoria = "Adolescente";
        } else if (edad < 60) {
            categoria = "Adulto";
        } else {
            categoria = "Adulto Mayor";
        }
        
        System.out.println("Su edad es: " + edad + " años");
        System.out.println("Categoría: " + categoria);
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite su edad: 35
 * 
 * === CATEGORÍA DE EDAD ===
 * Su edad es: 35 años
 * Categoría: Adulto
 */

File: JS/Tema3_Ejercicio6.js
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 6
 * ================================================
 * Clasificar edad en categorías.
 * 
 * EXPLICACIÓN:
 * - Rangos de edades
 * - Múltiples condiciones if-else if
 * - Clasificación por grupos etarios
 */

function clasificarEdad(edad) {
    if (edad < 0 || edad > 150) {
        return "Edad inválida";
    } else if (edad < 5) {
        return "Infante";
    } else if (edad < 12) {
        return "Niño";
    } else if (edad < 18) {
        return "Adolescente";
    } else if (edad < 65) {
        return "Adulto";
    } else {
        return "Adulto mayor";
    }
}

console.log("=== CLASIFICACIÓN POR EDAD ===");
const edades = [2, 8, 15, 25, 45, 70, 120, -5];

edades.forEach(edad => {
    console.log(`Edad ${edad}: ${clasificarEdad(edad)}`);
});

/*
 * SALIDA ESPERADA:
 * === CLASIFICACIÓN POR EDAD ===
 * Edad 2: Infante
 * Edad 8: Niño
 * Edad 15: Adolescente
 * Edad 25: Adulto
 * Edad 45: Adulto
 * Edad 70: Adulto mayor
 * Edad 120: Edad inválida
 * Edad -5: Edad inválida
 */
